google/cloud/pubsub/internal/subscription_batch_source.h

```diff
@@ -30,9 +31,6 @@ namespace cloud {
 namespace pubsub_internal {
 GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
 
-using BatchCallback =
```

Comment:
What is the advantage of replacing a `std::function` with a class that has a single `virtual operator()`?

Reply:
It will eventually have more. The main advantage is for tracing. So we can store the state of the spans around the callback

Reply:
Ack, yes there will be member functions to add events and links